plugins/pulp_rpm/plugins/migrations/0043_add_modulemd_modulemd-defaults.py

```diff
@@ -0,0 +1,40 @@
+import gzip
+import logging
+
+from pulp.plugins.loader import api as plugin_api
+from pulp.server.controllers import repository as repository_controller
+from pulp.server.db.model import Repository
+from pulp.server.db import connection
+from pulp_rpm.plugins.db.models import YumMetadataFile
+from pulp_rpm.plugins.importers.yum import modularity
+
+
+_logger = logging.getLogger('pulp_rpm.plugins.migrations.0043')
+
+
+def migrate(*args, **kwargs):
+
+    plugin_api.initialize()
+    db = connection.get_database()
+    repo_content = db['repo_content_units']
+    unit_ids = [unit['unit_id'] for unit in repo_content.find({"unit_type_id":
+                                                               "yum_repo_metadata_file"})]
+    metadatafiles = YumMetadataFile.objects(__raw__={'data_type': 'modules',
+                                                     '_id': {'$in': unit_ids}})
+    repos = set()
+    for file in metadatafiles:
+        repository = Repository.objects.get(repo_id=file.repo_id)
+        with gzip.open(file._storage_path, 'r') as fp:
+            working_dir = "/var/cache/pulp"
+            modulemds, defaults = modularity.load(fp, working_dir)
+            modularity.add_modulemds(repository, modulemds, working_dir)
+            modularity.add_defaults(repository, defaults, working_dir)
+            repository_controller.disassociate_units(repository, [file])
+            repository_controller.rebuild_content_unit_counts(repository)
+            repos.add(file.repo_id)
+    if repos:
+        _logger.info("The following repositories contain Module metadata files. These repositories "
+                     "need to be republished so they continue working after an orphan cleanup task "
+                     "runs.\n")
```

Comment:
We needed to republish after one puppet migration in the past and it was tackled like this:
https://github.com/pulp/pulp_puppet/blob/2-master/pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py#L55

The idea is that the list of repos to republish is available in a file for convenience and a later use/reference if needed.
What do you think?

Reply:
Thanks! I will do the same here.